artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageCursorProviderImpl.java

```diff
@@ -599,6 +600,18 @@ private long checkMinPage(Collection<PageSubscription> cursorList) {
 
    }
 
+   private void deliverIfNecessary(Collection<PageSubscription> cursorList, long minPage) {
+      for (PageSubscription cursor : cursorList) {
+         long firstPage = cursor.getFirstPage();
+         if (firstPage == minPage) {
+            if (cursor.getQueue().getMessageCount() == 0 && cursorList.size() > 1) {
+               cursor.getQueue().deliverAsync();
+               break;
```

Comment:
The `break` could be taken out, landing into `if (firstPage == minPage) { ..break; }`?
@wy96f @clebertsuconic how many cursors can have `cursor.getFirstPage() == minPage`?

Reply:
@wy96f  @clebertsuconic  Maybe more than one. I changed the code and added another test. Please review the code:)